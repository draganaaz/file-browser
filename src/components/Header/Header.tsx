import React from 'react';

interface HeaderProps {
  handleFilter: (value: string) => void;
}

const Header: React.FC<HeaderProps> = ({ handleFilter }) => {
  return (
    <header className="w-full p-4 bg-blue-500 text-white shadow-md flex flex-col items-center gap-4 md:flex-row md:justify-start md:items-center">
      <h1 className="text-2xl font-bold text-center md:text-left">
        File Browser
      </h1>

      <input
        type="text"
        placeholder="Filter files..."
        className="w-full max-w-xs p-2 rounded text-gray-700 md:w-auto"
        onChange={(e) => handleFilter(e.target.value)}
      />
    </header>
  );
};

export default Header;
