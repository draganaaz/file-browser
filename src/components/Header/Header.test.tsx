import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Header from './Header';

const mockHandleFilter = jest.fn();

describe('Header Component', () => {
  beforeEach(() => {
    mockHandleFilter.mockClear();
  });

  it('renders the header title', () => {
    render(<Header handleFilter={mockHandleFilter} />);
    const titleElement = screen.getByText('File Browser');
    expect(titleElement).toBeInTheDocument();
  });

  it('calls handleFilter when input changes', () => {
    render(<Header handleFilter={mockHandleFilter} />);

    const inputElement = screen.getByPlaceholderText('Filter files...');
    fireEvent.change(inputElement, { target: { value: 'test' } });

    expect(mockHandleFilter).toHaveBeenCalledTimes(1);
    expect(mockHandleFilter).toHaveBeenCalledWith('test');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Header handleFilter={mockHandleFilter} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
