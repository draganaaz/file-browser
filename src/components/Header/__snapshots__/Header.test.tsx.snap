// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="w-full p-4 bg-blue-500 text-white shadow-md flex flex-col items-center gap-4 md:flex-row md:justify-start md:items-center"
  >
    <h1
      class="text-2xl font-bold text-center md:text-left"
    >
      File Browser
    </h1>
    <input
      class="w-full max-w-xs p-2 rounded text-gray-700 md:w-auto"
      placeholder="Filter files..."
      type="text"
    />
  </header>
</DocumentFragment>
`;
